File: dotnet/xmsclient/Client/State/Listener/SpecificMethodFrameListener.cs
using System;
using OpenAMQ.Framing;
using OpenAMQ.XMS.Client.Protocol;

namespace OpenAMQ.XMS.Client.State.Listener
{
    public class SpecificMethodFrameListener : BlockingMethodFrameListener
    {
        private readonly Type _expectedClass;

        public SpecificMethodFrameListener(ushort channelId, Type expectedClass) : base(channelId)
        {        
            _expectedClass = expectedClass;
        }

        public bool processMethod(int channelId, AMQMethodBody frame)
        {
            return _expectedClass.IsInstanceOfType(frame);
        }
    }
}

File: java/src/org/openamq/client/state/listener/SpecificMethodFrameListener.java
package org.openamq.client.state.listener;

import org.openamq.client.protocol.BlockingMethodFrameListener;
import org.openamq.framing.AMQMethodBody;
import org.openamq.AMQException;

public class SpecificMethodFrameListener extends BlockingMethodFrameListener
{
    private final Class _expectedClass;

    public SpecificMethodFrameListener(int channelId, Class expectedClass)
    {
        super(channelId);
        _expectedClass = expectedClass;
    }

    public boolean processMethod(int channelId, AMQMethodBody frame) throws AMQException
    {
        return _expectedClass.isInstance(frame);
    }
}
